distogram/__init__.py

```diff
@@ -1,240 +1,203 @@
+__author__ = """Romain Picard"""
+__email__ = 'romain.picard@oakbits.com'
+__version__ = '1.7.0'
+
 import math
-from functools import reduce
 from bisect import bisect_left
+from functools import reduce
+from operator import itemgetter
+from typing import List
+from typing import Optional
+from typing import Tuple
 
-__author__ = """Romain Picard"""
-__email__ = 'romain.picard@oakbits.com'
-__version__ = '1.6.0'
+EPSILON = 1e-5
+Bin = Tuple[float, int]
 
 
 # bins is a tuple of (cut point, count)
 class Distogram(object):
-    '''Compressed representation of a distribution
-    '''
+    """ Compressed representation of a distribution.
+    """
     __slots__ = 'bin_count', 'bins', 'min', 'max', 'diffs', 'min_diff', 'weighted_diff'
 
-    def __init__(self, bin_count=100, weighted_diff=False):
-        '''Creates a new Distogram object
+    def __init__(self, bin_count: int = 100, weighted_diff: bool = False):
+        """ Creates a new Distogram object
 
         Args:
             bin_count: [Optional] the number of bins to use.
+            weighted_diff: [Optional] Whether to use weighted bin sizes.
 
         Returns:
             A Distogram object.
-        '''
```

Comment:
PEP8 says that DocStrings use double quotes rather than single quotes. I personally don't have a preference. My IDE just made this change automatically and we can always revert it.